com.microsoft.mrtk.tools/SubsystemWizard/SubsystemGenerator.cs

```diff
@@ -0,0 +1,595 @@
+        /// source file.
+        /// </summary>
+        public bool CreateConfiguration
+        {
+            get => createConfiguration;
+            set => createConfiguration = value;
+        }
+
+        [SerializeField]
+        private string organizationName = string.Empty; // DefaultOrganizationName;
+
+        /// <summary>
+        /// The name of the entity which is creating / releasing the subsystem.
+        /// </summary>
+        public string OrganizationName
+        {
+            get => organizationName;
+            set => organizationName = value;
+        }
+
+        /// <summary>
+        /// Name of configuration class, if enabled by <see cref="CreateConfiguration"/>
+        /// to create for new subystem.
+        /// </summary>
+        public string ConfigurationName => CreateConfiguration ? $"{BaseClassName}Config" : "BaseSubsystemConfig"; 
+
+        /// <summary>
+        /// Name of descriptor class to create for new subystem.
+        /// </summary>
+        public string DescriptorName => $"{BaseClassName}Descriptor";
+
+        /// <summary>
+        /// The name that will be displayed in project settings.
+        /// </summary>
+        public string DisplayName => $"{OrganizationName} {BaseClassName}";
+
+        /// <summary>
+        /// Name of interface to create for new subystem.
+        /// </summary>
+        public string InterfaceName => $"I{BaseClassName}";
+
+        private bool dontCreateDescriptor = false;
+
+        /// <summary>
+        /// Indicates if the user wishes to skip the creation of the subsystem
+        /// descriptor source code file.
+        /// </summary>
+        public bool DontCreateDescriptor
+        {
+            get => dontCreateDescriptor;
+            set => dontCreateDescriptor = value;
+        }
+
+        private bool dontCreateInterface = false;
+
+        /// <summary>
+        /// Indicates if the user wishes to skip the creation of the subsystem
+        /// interface source code file.
+        /// </summary>
+        public bool DontCreateInterface
+        {
+            get => dontCreateInterface;
+            set => dontCreateInterface = value;
+        }
+
+        private bool dontCreateClass = false;
+
+        /// <summary>
+        /// Indicates if the user wishes to skip the creation of the subsystem
+        /// class source code file.
+        /// </summary>
+        public bool DontCreateBaseClass
+        {
+            get => dontCreateClass;
+            set => dontCreateClass = value;
+        }
+
+        private bool dontCreateImplementationClass = false;
+
+        /// <summary>
+        /// Indicates if the user wishes to skip the creation of the subsystem
+        /// descriptor source code file.
+        /// </summary>
+        public bool DontCreateImplementationClass
+        {
+            get => dontCreateImplementationClass;
+            set => dontCreateImplementationClass = value;
+        }
```

Comment:
super nit: could these be auto-properties?

Reply:
yep. just a habit when writing mrtk code 😄

Reply:
updated